frontend/src/components/namespace/CreateNamespaceButton.tsx

```diff
@@ -0,0 +1,168 @@
+import {
+  Box,
+  Button,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  TextField,
+} from '@mui/material';
+import React, { useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useDispatch } from 'react-redux';
+import { getCluster } from '../../lib/cluster';
+import Namespace from '../../lib/k8s/namespace';
+import { clusterAction } from '../../redux/clusterActionSlice';
+import { EventStatus, HeadlampEventType, useEventCallback } from '../../redux/headlampEventSlice';
+import { ActionButton, AuthVisible } from '../common';
+
+export default function CreateNamespaceButton() {
+  const { t } = useTranslation(['glossary', 'translation']);
+  const [namespaceName, setNamespaceName] = useState('');
+  const [isValidNamespaceName, setIsValidNamespaceName] = useState(false);
+  const [nameHelperMessage, setNameHelperMessage] = useState('');
+  const [namespaceDialogOpen, setNamespaceDialogOpen] = useState(false);
+  const dispatchCreateEvent = useEventCallback(HeadlampEventType.CREATE_RESOURCE);
+  const dispatch = useDispatch();
+
+  function createNewNamespace() {
+    const clusterData = getCluster();
+    const newNamespaceData = {
+      apiVersion: 'v1',
+      kind: 'Namespace',
+      metadata: {
+        name: namespaceName,
+      },
+    };
+
+    const newNamespaceName = newNamespaceData.metadata.name;
+
+    async function namespaceRequest() {
+      try {
+        const response = await Namespace.apiEndpoint.post(newNamespaceData, {}, clusterData || '');
+        setNamespaceName('');
+        return response;
+      } catch (error: any) {
+        const statusCode = error?.status;
+        console.error('Error creating namespace:', error);
+        if (statusCode === 409) {
+          setNamespaceDialogOpen(true);
+          setIsValidNamespaceName(false);
+          setNameHelperMessage(t('translation|A namespace with this name already exists.'));
+        }
+        throw error;
+      }
+    }
+
+    setNamespaceDialogOpen(false);
+
+    dispatch(
+      clusterAction(() => namespaceRequest(), {
+        startMessage: t('translation|Applying {{  newItemName  }}…', {
+          newItemName: newNamespaceName,
+        }),
+        cancelledMessage: t('translation|Cancelled applying  {{  newItemName  }}.', {
+          newItemName: newNamespaceName,
+        }),
+        successMessage: t('translation|Applied {{ newItemName }}.', {
+          newItemName: newNamespaceName,
+        }),
+        errorMessage: t('translation|Failed to create {{ kind }} {{ name }}.', {
+          kind: 'namespace',
+          name: newNamespaceName,
+        }),
+        cancelCallback: () => {
+          setNamespaceDialogOpen(true);
+        },
+      })
+    );
+  }
+
+  useEffect(() => {
+    setIsValidNamespaceName(Namespace.isValidNamespaceFormat(namespaceName));
+
+    if (!Namespace.isValidNamespaceFormat(namespaceName)) {
```

Comment:
These 2 lines call the same function, please set the result of isValidNamespaceFormat to a const and reuse it.

Reply:
done!